closed/make/Docs.gmk

```diff
@@ -20,4 +20,115 @@
+# for files found before they are renamed, causing build errors.  
+
+OPENJDK_SUN_MGMT_SRCDIR = $(OPENJDK_JCL_TOPDIR)/jdk.management/share/classes/jdk.management/com/sun/management
+OPENJ9_SUN_MGMT_SRCDIR = $(OPENJ9_JCL_TOPDIR)/jdk.management/share/classes/com/sun/management
+
+JDK_API_CUSTOM_TARGETS += openj9-docs-set-javadoc-bottom openj9-docs-restore-files
+
+# Code should only be executed when generating the docs-jdk-api-javadoc target.
+# If executed for any other targets the renames would be done but the target
+# to restore the files would not be executed.
+
+ifneq (,$(findstring docs-jdk-api-javadoc,$(MAKECMDGOALS)))
+  # First rename the files we want to generate javadoc for, and to keep package-info.java which may be present from a previous run.
+  CMD_OUTPUT = $(shell if [ -f $(OPENJDK_SUN_MGMT_SRCDIR)/GarbageCollectorMXBean.java ]; then $(MV) $(OPENJDK_SUN_MGMT_SRCDIR)/GarbageCollectorMXBean.java $(OPENJDK_SUN_MGMT_SRCDIR)/GarbageCollectorMXBean.javaz; fi)
+  CMD_OUTPUT = $(shell if [ -f $(OPENJDK_SUN_MGMT_SRCDIR)/OperatingSystemMXBean.java ]; then $(MV) $(OPENJDK_SUN_MGMT_SRCDIR)/OperatingSystemMXBean.java $(OPENJDK_SUN_MGMT_SRCDIR)/OperatingSystemMXBean.javaz; fi)
+  CMD_OUTPUT = $(shell if [ -f $(OPENJDK_SUN_MGMT_SRCDIR)/UnixOperatingSystemMXBean.java ]; then $(MV) $(OPENJDK_SUN_MGMT_SRCDIR)/UnixOperatingSystemMXBean.java $(OPENJDK_SUN_MGMT_SRCDIR)/UnixOperatingSystemMXBean.javaz; fi)
+  CMD_OUTPUT = $(shell if [ -f $(OPENJDK_SUN_MGMT_SRCDIR)/package-info.java ]; then $(MV) $(OPENJDK_SUN_MGMT_SRCDIR)/package-info.java $(OPENJDK_SUN_MGMT_SRCDIR)/package-info.javaz; fi)
+  # Then rename any remaining files we don't want to generate javadoc for
+  # This finds all files ending in .java and then renames each one with an 'x' added add its last character.
+  CMD_OUTPUT = $(shell find $(OPENJDK_SUN_MGMT_SRCDIR) -maxdepth 1 -type 'f' -name "*.java" | xargs -I {} sh -c 'set -x; $(MV) $$1 $${1}x; set +x' -- {})
+  # HotSpotDiagnosticMXBean.java is just a stub class in OpenJ9, so don't generate javadoc for it.
+  CMD_OUTPUT = $(shell if [ -f $(OPENJ9_SUN_MGMT_SRCDIR)/HotSpotDiagnosticMXBean.java ]; then $(MV) $(OPENJ9_SUN_MGMT_SRCDIR)/HotSpotDiagnosticMXBean.java $(OPENJ9_SUN_MGMT_SRCDIR)/HotSpotDiagnosticMXBean.javax; fi)
+  # Finally restore the .java extension of the files we do want to generate javadoc for and the package-info.java file if present
+  CMD_OUTPUT = $(shell find $(OPENJDK_SUN_MGMT_SRCDIR) -maxdepth 1 -type 'f' -name "*.javaz" | xargs -I {} sh -c 'set -x; $(MV) $$1 $${1%?}; set +x' -- {})
+endif
+
+# Target to change the web page footers to the required values.
+# This ensures that openj9 and openjdk classes contain the correct copyrights etc.
+openj9-docs-set-javadoc-bottom:
+	# Set footer for openj9 modules, packages and classes
+	cd $(DOCS_OUTPUTDIR)/api; find . -type 'f' | grep "/openj9" | grep ".html" | xargs -I {} $(OPENJ9_SUBST_COMMAND) {}
```

Comment:
`grep /openj9 | grep .html` matches too many paths;
I think you want something more like `grep '/openj9\\..*\\.html$'.

Reply:
The code is saying "find all files under DOCS_OUTPUTDIR/api which contain '/openj9' in their path (gets all files in an openj9xxxx sub-directory) and also contain .html ((probably) limits the files to those with an .html extension - this will include all those created by javadoc), and then substitute any occurrences of JAVADOC_BOTTOM in those files with the correct html footer for the code in question".  If there is an html (or other file) accidentally picked up by the grep which is not a javadoc created file I assume it will not contain the test JAVADOC_BOTTOM so the sed will be a noop.  My feeling is that keeping the filters simple will make it more easily understood and maintainable going forward.

Reply:
I suggest you at least replace the second `grep` with the extra predicate `-name '*.html'` in the `find` command: it's still clear and won't match arbitrary paths containing 'html'.

Reply:
-name "*.html" added (now in closed/make/Docs-post.gmk).